variables.tf

```diff
@@ -93,6 +93,20 @@ variable "api_server_authorized_ip_ranges" {
   default     = null
 }
 
+variable "automatic_channel_upgrade" {
+  type        = string
+  default     = null
+  description = "(Optional) The upgrade channel for this Kubernetes Cluster. Possible values are  `patch`, `rapid`, `node-image` and `stable`. By default automatic-upgrades are turned off. Note that you cannot use the `patch` upgrade channel and still specify the patch version using `kubernetes_version`. See [the documentation](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-cluster) for more information"
+  validation {
+    condition = (
+      var.automatic_channel_upgrade == null || (
```

Comment:
Since the [Terraform's binary operator is not short-circuit](https://github.com/hashicorp/terraform/issues/24128), we would have an issue when `var.automatic_channel_upgrade` is `null`, though `var.automatic_channel_upgrade == null` is `true`, Terraform would continue to evaluate the value of `contains(["patch", "stable", "rapid", "node-image"], var.automatic_channel_upgrade)`, which equals to `contains(["patch", "stable", "rapid", "node-image"], null)`, and that would caused an error, the `contains` function dosen't accept `null` as parameter.

I would recommend the following expression:

```hcl
var.automatic_channel_upgrade == null ? true : (contains(["patch", "stable", "rapid", "node-image"], var.automatic_channel_upgrade))
```

Reply:
Good point, I've read about this and then forgot it again. Switched to your snippet.